release-notes/0.33.0.rst

```diff
@@ -0,0 +1,27 @@
+0.33.0 (2024-11-07)
+===================
+
+New Features
```

Comment:
We use sentence case for headers (so the words after the first word wouldn't be upper-case), but I hesitate to suggest fixing this if the release notes have been done this way in many past iterations, meaning a break in consistency ... I leave it to the runtime team to make that call.